analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -199,25 +209,25 @@ private void addOrUpdateActivity(UserAnonymizedEntityHolder userAnonymizedHolder
 			ActivityPayload nextDayPayload = ActivityPayload.copyFromStartTime(payload,
 					TimeUtil.getStartOfDay(payload.userAnonymized.getTimeZone(), payload.endTime));
 
-			addOrUpdateDayTruncatedActivity(userAnonymizedHolder, truncatedPayload, matchingGoal);
-			addOrUpdateDayTruncatedActivity(userAnonymizedHolder, nextDayPayload, matchingGoal);
+			addOrUpdateDayTruncatedActivity(truncatedPayload, matchingGoal, userAnonymizedEntityHolder);
+			addOrUpdateDayTruncatedActivity(nextDayPayload, matchingGoal, userAnonymizedEntityHolder);
 		}
 		else
 		{
-			addOrUpdateDayTruncatedActivity(userAnonymizedHolder, payload, matchingGoal);
+			addOrUpdateDayTruncatedActivity(payload, matchingGoal, userAnonymizedEntityHolder);
 		}
 	}
 
-	private void addOrUpdateDayTruncatedActivity(UserAnonymizedEntityHolder userAnonymizedHolder, ActivityPayload payload,
-			GoalDto matchingGoal)
+	private void addOrUpdateDayTruncatedActivity(ActivityPayload payload, GoalDto matchingGoal,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This method has 24 lines, which is greater than the 20 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://build.dev.yona.nu:9000/coding_rules#rule_key=squid%3AS138)